### PR TITLE
[FIX] website_sale_coupon: prevent race condition

### PR DESCRIPTION
**Pull request by rdeodoo**
2 lasts steps are:
  - Click on go to checkout
  - Ensure checkout total is correct

But the last condition was already true before landing to the checkout, thus
the last step would succeed while the page is transitionning to the checkout.

The following would happen every time:
  - Click on go to checkout
  - Last step succeed directly after the previous step click on Go to checkout
Then, the following would happen rarely:
  - The tour is considered successful
  - The checkout page is loaded
  - Some GET requests are performed
  - The tour is killed (cookies & local storage)
  - Some other GET requests are crashing, such as:
    `Error loading file http://127.0.0.1:8069/web/webclient/locale/en_US`

The fix simply wait for the checkout page to be loaded as it was supposed to at
first, before checking the last step.

Forward-port of https://github.com/odoo/enterprise/pull/8322

X-original-commit: https://github.com/odoo/enterprise/commit/67686b697c866a5e5ad141a85b3589f1f1598d73
(coupon was moved from enterprise to community in Odoo saas-12.3)

### CODE REVIEW

File: addons/delivery/models/delivery_carrier.py [hunk 1]
```diff
@@ -258,7 +258,8 @@ def fixed_rate_shipment(self, order):
                     'warning_message': False}
         price = self.fixed_price
         if self.company_id and self.company_id.currency_id.id != order.currency_id.id:
-            price = self.env['res.currency']._compute(self.company_id.currency_id, order.currency_id, price)
+            date = order.date_order or fields.Date.today()
+            price = self.env['res.currency']._convert(price, order.currency_id, self.company_id, date)
```
Comment (rdeodoo) [inline addons/delivery/models/delivery_carrier.py]: @d-fence There was still a call to a deprecated method.
This tour is the only one going through checkout payment page when `website_sale_delivery` is installed.
I am not sure about `self.company_id` vs `order.company_id` as third parameter tho.

Comment (d-fence) [reply addons/delivery/models/delivery_carrier.py]: As the condition is checking if self.company_id is truthy, I guess that we want to retrieve the conversion rate from the self.company_id. It seems logical to get the conversion rate from the company of the delivery method if set.

Comment (rdeodoo) [reply addons/delivery/models/delivery_carrier.py]: Do you want to validate this commit by someone else? Otherwise you can r+ since I can't :yum:

**Comment by d-fence** [comment]
robodoo r+

**Comment by rdeodoo** [comment]
@d-fence rebase-ff

**Comment by d-fence** [comment]
oops, your private robodoo proxy failed :smile: 
robodoo rebase-ff

**Pull request closed without merging**